### ISSUE TITLE
how to prevent routing when selecting from dropdown?

### ISSUE THREAD

**Issue by schristley** [xcash/bootstrap-autocomplete#82]
**Describe the bug**
For an autoSelect, when an item is picked from the dropdown, the browser routes to `!`. I tried preventing, like can be done with buttons, with this code but that doesn't seem to work:

```
        'autocomplete.select': function(evt, item) {
            evt.preventDefault();
        }
```

**Expected behavior**
I don't expect the route to change. I just trap the event and update values myself.

**Desktop (please complete the following information):**
 - OS: Mac
 - Browser safari
 - Version 13.1.2

**Additional context**
The website is a javascript single-page app, so it manages all the routing, and all form submits are handled through ajax and not with the default browser action.

Using bootstrap V4. With the other bootstrap dropdowns in our app, we do not use `a href` for the dropdown items.

**Comment by schristley** [comment]
Well, it is different, but it still has the same problem. Now instead of routing to `!`, it routes to the `/` index page. Eliminating the `href` completely solves the problem. I guess though you need `href` for some sort of styling? I don't see a difference with or without it, but maybe I'm missing something.

```
diff --git a/src/dropdownV4.ts b/src/dropdownV4.ts
index ba6184c..1c17daf 100644
--- a/src/dropdownV4.ts
+++ b/src/dropdownV4.ts
@@ -208,8 +208,8 @@ export class DropdownV4 {
         const disabledItem = itemFormatted.disabled;
 
         const li = $('<a >');
-        li.attr('href', '')
-          .addClass('dropdown-item')
+        // li.attr('href', '')
+          li.addClass('dropdown-item')
           .css({ 'overflow': 'hidden', 'text-overflow': 'ellipsis' })
           .html(itemHtml)
           .data('item', item);
@@ -229,8 +229,8 @@ export class DropdownV4 {
       } else {
         // show no results message
         const li = $('<a >');
-        li.attr('href', '')
-          .addClass('dropdown-item disabled')
+        // li.attr('href', '')
+          li.addClass('dropdown-item disabled')
           .html(this.noResultsText);
```

Another idea is to add a `preventHref` setting, which allows the user to decide. This is a bit more changes because the setting will need to be passed as another parameter to the dropdown constructor, but not too bad. I could give you a PR if you'd like?

**Comment by xcash** [comment]
Hi @schristley 
well this is indeed strange as I prevented default ( https://github.com/xcash/bootstrap-autocomplete/blob/master/src/dropdownV4.ts#L55 ) and I don't see this happening in the test page.
Are you using Vue?
I think I could safely drop href but I normally tend to fully understand what an issue really is before applying a workaround. 
Let me know :)

**Comment by schristley** [comment]
> Are you using Vue?

No, we use Marionette/Backbone, and I tend to develop with the Safari browser. I don't think they are the problem though, but I admit that I'm not completely sure either. We had been using bootstrap 3 for our website but now we are rewriting with bootstrap 4. I have a feeling that it's a bootstrap 4 issue. For example, here's a standard dropdown, cut/paste from the [docs](https://getbootstrap.com/docs/4.5/components/dropdowns/#single-button)

```
<div class="dropdown">
  <button class="btn btn-secondary dropdown-toggle" type="button" id="dropdownMenuButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
    Dropdown button
  </button>
  <div class="dropdown-menu" aria-labelledby="dropdownMenuButton">
    <a class="dropdown-item" href="#">Action</a>
    <a class="dropdown-item" href="#">Another action</a>
    <a class="dropdown-item" href="#">Something else here</a>
  </div>
</div>
```

It will route when clicking an item. Now if I try to prevent default with the click event, it still routes! The only way I've been able to prevent the automatic routing is by eliminating the `href`

```
        'click .dropdown-item': function(e) {
            console.log('click');
            e.preventDefault();
        }
```

I found a stackoverflow that suggested doing something like this, this does prevent the routing but it also breaks the dropdown in that it doesn't close when the item is clicked.

```
        'click .dropdown-item': function(e) {
            console.log('click');
            e.preventDefault();
            e.stopPropagation();
        }
```

So maybe this is a bug with bootstrap v4? It's interesting that you say it works for you... I was using v4.3.1, and I upgraded to the latest v4.5.2, but still the same behavior.

The bootstrap v4 dropdown docs don't seem to really talk about handling the click.

**Comment by xcash** [comment]
I can't test in Safari here (I develop on linux and I test in chrome/firefox). I assume you see the same issue even in a browser other than Safari.. ?

**Comment by schristley** [comment]
Yes, I just tested with Chrome and the behavior is the same.

**Comment by xcash** [comment]
Ok then.... let's remove the href... :D
